### ISSUE TITLE
Ability to generate rich models from Entity concept

### ISSUE THREAD

**Issue by Halky** [Rhetos/Rhetos#145]
**Is your feature request related to a problem? Please describe.**
At the moment, Rhetos generates anemic models (simple DTOs) which leads to poor encapsulation in both inside Rhetos DSL scripts and applications which are consuming generated models.
Typical symptoms of this problem are a lot of helper classes or access logic to computed data (like state of record in event sourcing scenario) spread and duplicated all over the application.

**Describe the solution you'd like**
Ability to enrich Entites with instanced, static and extension methods. It would be also very helpful to add "computed" properties to model along with implementation of theirs setters and getters.
Example of "computed" property would be state of an object which is computed from different table(s). "Computed" property would not created an additional column in the database, it would live only in the model of an entity (similar to View-Model pattern).

E.g.
```
Entity Person
{
     ShortString Name;
     ShortString LastName;
     Bool IsMarried //"computed" property
     {
          get { repository.Schema1.Marrages.Query(m => m.ID == this.ID).Any(); }
     }
     String GetFullName() //instanced method on Person object
     {
         return $"{this.Name} {this.LastName}";
     }
}
```

**Comment by bantolov** [comment]
I agree that the we need better encapsulation, especially to improve code readability and discoverability.
We need more discussion on this topic.

This issue is somewhat related to a roadmap goal [Complex web API methods](https://github.com/Rhetos/Rhetos/wiki/Rhetos-platform-roadmap#3-complex-web-api-methods) and issue #50. It might be split into multiple more specific issues.

Just for reference, here are some currently available approaches for implementing the examples above:

With LINQ:

```c
QueryableExtension PersonInfo1 Demo.Person
  '(source, repository) => source.Select(p => new Common.Queryable.Demo_PersonInfo1
    {
        ID = p.ID,
        IsMarried = repository.Demo.Marriage.Subquery.Any(m => m.PersonID == p.ID),
        FullName = p.Name + " " + p.LastName
    })
  '
{
    Bool IsMarried;
    ShortString FullName;
}
```

With SQL:

```c
SqlQueryable PersonInfo2
  "
    SELECT
        p.ID,
        IsMarried = CASE WHEN EXISTS
            (SELECT * FROM Demo.Marriage m WHERE m.PersonID = p.ID)
            THEN CAST(1 AS BIT) ELSE CAST (0 AS BIT) END,
        FullName = p.Name + ' ' + p.LastName
    FROM
        Demo.Person p
  "
{
    Extends Demo.Person;
    Bool IsMarried;
    ShortString FullName;
}
```

Additionally, a Browse should be added if we want to combine the computed properties from different objects, but it will only help with simplest situations until more work is done on "Complex web API methods".
